superset-frontend/src/views/dashboardList/DashboardList.tsx

```diff
@@ -251,8 +250,33 @@ class DashboardList extends React.PureComponent<Props, State> {
     return Boolean(this.state.permissions.find(p => p === perm));
   };
 
-  handleDashboardEdit = ({ id }: { id: number }) => {
-    window.location.assign(`/dashboard/edit/${id}`);
+  openDashboardEditModal = (dashboard: Dashboard) => {
+    this.setState({
+      dashboardToEdit: dashboard,
+    });
+  };
+
+  handleDashboardEdit = (edits: any, original: Dashboard) => {
+    this.setState({ loading: true });
+    return SupersetClient.get({
+      endpoint: `/api/v1/dashboard/${original.id}`,
```

Comment:
Is there a reason we're making this call instead of just updating from `edits`?

Reply:
I think there was some field that wasn't returned from the `PUT` endpoint. I wasn't sure how to add that field and eventually just did this to make it work. If you think it's worth the effort I'll take another pass at it, otherwise I can add a comment documenting why it's there.

Reply:
happy to merge this, this shouldn't affect perf much. However, we should figure out how to configure the PUT response

Reply:
might be `edit_columns`

Reply:
I remember this now. The data they return uses the `edit_model_schema` Marshmallow schema which is the same as the one used to parse incoming edited fields from the client. IMO a better design would be to return the entire object as it would from the `GET` endpoint, but that's a larger scale API design decision.

Reply:
In this particular case, the impact is that `slug` can be changed, which changes the `url`, but only `slug` is returned from the endpoint.